### PR TITLE
chore(strict-null): add strict null check migration tools to repo

### PR DESCRIPTION
**Pull request by dbjorge**
#### Description of changes

This integrates the strict null checking tools that I migrated from VS Code's as an experiment in https://github.com/dbjorge/aiweb-strict-null-check-migration-tools into the `/tools/` folder of our repo, to make it easier for others to use them.

As part of the migration, I eliminated the separate `package.json` machinery for the tools and had them just re-use the same `node_modules` as the main project (in particular, re-using the correct typescript version). I also added entry point run-scripts for the tools to our root-level package.json, so the source for the tools should be an implementation detail for most purposes of actually running null check migration work; usage is just `yarn null:find` and `yarn null:autoadd` from the root, to match the previously-added `yarn null:check`.

I've updated #2869 with instructions for how to use these scripts to contribute to the strict null check migration work.

#### Pull request checklist

- [x] Addresses an existing issue: tools to help address #2869 
- [x] Ran `yarn fastpass`
- [n/a] Added/updated relevant unit test(s) (and ran `yarn test`)
- [n/a] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: tools/strict-null-checks/auto-add.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                    fs.writeFileSync(tsconfigPath, JSON.stringify(originalConfig, null, '\t'));
+                }
+
+                child.stdout.removeListener('data', listener);
+                resolve();
+            }
+        };
+        child.stdout.on('data', listener);
+
+        fs.writeFileSync(tsconfigPath, JSON.stringify(newConfig, null, '    '));
```
Comment (karanbirsingh) [inline tools/strict-null-checks/auto-add.js:60]: how does this work in concert with line 48? It seems like only one would be needed, but I might be missing something

File: tools/strict-null-checks/auto-add.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        // Config on accept
+        const newConfig = Object.assign({}, originalConfig);
+        newConfig.files = Array.from(
+            new Set(originalConfig.files.concat('./' + relativeFilePath).sort()),
+        );
+
+        const listener = data => {
+            const textOut = data.toString();
+            // console.log('  ' + textOut);
```
Comment (karanbirsingh) [inline tools/strict-null-checks/auto-add.js:42]: can remove(?)

File: tools/strict-null-checks/auto-add.js (hunk 1 above)
Comment (dbjorge) [reply tools/strict-null-checks/auto-add.js:58]: Agreed, 48 is unnecessary

File: tools/strict-null-checks/auto-add.js (hunk 2 above)
Comment (dbjorge) [reply tools/strict-null-checks/auto-add.js]: Yep

**Pull request merged**